### ISSUE TITLE
Still importing items even though type set in ExcludedEntityTypes

### ISSUE THREAD

**Issue by Rockerby** [umbraco/Umbraco.Deploy.Issues#232]
After setting `ExcludedEntityTypes: [ "dictionary-item" ]` the options for dictionary items was removed from the deploy dashboard area, but running `Update Umbraco Schema From Data Files` still tried to import them.

The specific scenario I was having was conflicting UDA files against pre-existing items (see below error) so my thought was to just exclude them from the deploy task to just ignore all dictionary items.

```
{
  "Date": "2024-08-06T09:04:54.585392+00:00",
  "Message": "Some artifacts collide on unique identifiers.\r\nThis means that they have different Udis, yet\r\nthey refer to the same unique Umbraco object\r\nand therefore cannot be processed.\r\n---------------------------------------------\r\nCollisions for entity type \"dictionary-item\": \r\n  Collisions for unique identifier \"XXXX\":\r\n    UdaFile: ~/umbraco/Deploy/Revision/dictionary-item__799133c706c54e30875ef023e734f9e5.uda\r\n    UdaFile: ~/umbraco/Deploy/Revision/dictionary-item__a9c9a12ed81345d79770f89fdcab1842.uda\r\n",
  "Ex": {
    "Report": {
      "HasCollisions": true,
      "TotalCollisions": 2,
      "Collisions": {
        "dictionary-item": {
          "XXXX": [
            "UdaFile: ~/umbraco/Deploy/Revision/dictionary-item__799133c706c54e30875ef023e734f9e5.uda",
            "UdaFile: ~/umbraco/Deploy/Revision/dictionary-item__a9c9a12ed81345d79770f89fdcab1842.uda"
          ]
        }
      }
    },
    "EnvironmentName": "source",
    "HasMeaningForUi": false,
    "Message": "Some artifacts collide on unique identifiers.\r\nThis means that they have different Udis, yet\r\nthey refer to the same unique Umbraco object\r\nand therefore cannot be processed.\r\n---------------------------------------------\r\nCollisions for entity type \"dictionary-item\": \r\n  Collisions for unique identifier \"Winter 2025\":\r\n    UdaFile: ~/umbraco/Deploy/Revision/dictionary-item__799133c706c54e30875ef023e734f9e5.uda\r\n    UdaFile: ~/umbraco/Deploy/Revision/dictionary-item__a9c9a12ed81345d79770f89fdcab1842.uda\r\n",
    "Data": {},
    "InnerException": null,
    "HelpLink": null,
    "Source": "Umbraco.Deploy.Infrastructure",
    "HResult": -2146233088,
    "StackTrace": "   at Umbraco.Deploy.Infrastructure.Work.WorkItems.DiskReadWorkItem.ExecuteAsyncSub(IWorkContext context, List`1 resume, CancellationToken token)\r\n   at Umbraco.Deploy.Infrastructure.Work.WorkItems.DiskReadWorkItem.ExecuteAsync(IWorkContext context, CancellationToken token)"
  },
  "Sender": "Umbraco.Deploy.Infrastructure.Work.WorkItems.DiskReadWorkItem"
}
```

I'm referencing the docs that can be found https://docs.umbraco.com/umbraco-deploy/getting-started/deploy-settings#excludedentitytypes which curiously says the recommendation is to not set this (but no explanation!)

Thanks ✌️
Richard

Reproduction
------------

### Bug summary

After setting the exclude setting, the dictionary item UDAs should be ignored.

### Steps to reproduce

On a site running deploy, create a dictionary item which will create the UDA. Set the setting `ExcludedEntityTypes: [ "dictionary-item" ]`, restart and this should stop the dictionary item UDAs from being recognised.

### Expected result

For all existing dictionary items to stay intact, no new dictionary item UDA files to be generated and all existing dictionary item UDA files to be ignored completely from future deploys.

### Actual result

Dictionary item UDAs still seem to be being processed after the setting is set.

---
_This item has been added to our backlog AB#42845_

**Comment by AndyButland** [comment]
Thanks for reporting @Rockerby.  I agree what you've described seems like it should be the expected behaviour.  Having had a look though, it seems this setting is only taken account of for backoffice transfer and restore operations, and not read/write to disk.  That surprised me to be honest.  The setting has been around for a while and I thought it worked how you expected.

It's fairly straightforward to extend it to disk operations too, so I'll do that and have someone review to see if there's anything I've missed and there's a good reason for this setting not to apply to disk operations.  All being well, we'll include this updated behaviour in a coming update.

> which curiously says the recommendation is to not set this (but no explanation!)

Again, this has been there a while and I think it's more just to make sure you don't do this without considering the consequences.  For something with a hard dependency - e.g. a data type and a document type - you can see that excluding one and not the other isn't going to lead to things working well. 

But for your case with dictionary items, that whilst are used in other entities like content, don't have a "hard" dependency in like the previous example, it would seem reasonable to do.

**Comment by Rockerby** [comment]
Thanks for the update on this @AndyButland.

I did put this up in Discord first and Seb pointed me in the direction of TransferDictionaryAsContent, which I could set to false and remove any uda files to essentially do what I needed to do.

Totally agree with the comments around further reaching impact of disabling, say, data types. If ExcludedEntityTypes doesn't have any other functionality than dashboard, another thought would be to leave it as is but update the documentation and advise TransferDictionaryAsContent is used for ignoring dictionary items?
